src/Meziantou.Analyzer/Rules/AvoidUsingRedundantElseFixer.cs

```diff
@@ -19,22 +19,39 @@ public sealed class AvoidUsingRedundantElseFixer : CodeFixProvider
     {
         public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(RuleIdentifiers.AvoidUsingRedundantElse);
 
-        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
-
-        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        /// <inheritdoc/>
+        public override FixAllProvider GetFixAllProvider()
         {
-            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
-            var nodeToFix = root.FindNode(context.Span, getInnermostNodeForTie: true);
-            if (nodeToFix == null)
-                return;
+            // We can't use WellKnownFixAllProviders.BatchFixer here, as we need to fix diagnostics per document in reverse order
+            return AvoidUsingRedundantElseFixAllProvider.Instance;
+        }
 
+        public override Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
             var title = "Remove redundant else";
             var codeAction = CodeAction.Create(
                 title,
-                ct => RemoveRedundantElse(context.Document, nodeToFix, ct),
+                ct => RemoveRedundantElseClausesInDocument(context.Document, context.Diagnostics, ct),
                 equivalenceKey: title);
 
             context.RegisterCodeFix(codeAction, context.Diagnostics);
+
+            return Task.CompletedTask;
+        }
+
+        internal static async Task<Document> RemoveRedundantElseClausesInDocument(Document document, ImmutableArray<Diagnostic> diagnostics, CancellationToken cancellationToken)
+        {
+            foreach (var diagnostic in diagnostics.Reverse())
```

Comment:
Are you sure the diagnostic are sorted? Or should you use OrderByDescending(Location)?

Reply:
You're right: I made an assumption, based on my test results.
Using OrderByDescending instead...